jazelle/utils/lockfile.js

```diff
@@ -467,4 +476,45 @@ const isBetterVersion = (version, range, graph, key) => {
   );
 };
 
+const getRegistry = async dir => {
+  const registry = await exec('yarn config get registry', {cwd: dir});
+  return registry ? registry.trim() : '';
+};
+
+const writeYarnRc = async (cwd, packageDir) => {
+  const yarnrc = ['"--install.frozen-lockfile" false'];
+  const registry = await getRegistry(packageDir);
+  if (registry) {
+    yarnrc.push(`--registry "${registry}"`);
+  }
+  await write(`${cwd}/.yarnrc`, yarnrc.join('\n'), 'utf8');
```

Comment:
Will there be one? This code is called after a temporary directory is generated to install missing or transitive dependencies.

So in the current context of where it is called there won't be one. But I can see the argument that this is a standalone function and so it could be called from any context. This was broken out into its own function really just for DRY'ness but I can add a check to handle a pre-existing .yarnrc if you feel strongly about it.

Reply:
Oh, I see, cwd here refers to the tmp directory. I thought it referred to the project root. In this case, there's no risk of clobbering a yarnrc, so this should be fine as is